homeassistant/components/solax/__init__.py

```diff
@@ -21,19 +37,29 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             entry.data[CONF_PORT],
             entry.data[CONF_PASSWORD],
         )
-        await api.get_data()
     except Exception as err:
         raise ConfigEntryNotReady from err
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = api
+    async def _async_update() -> InverterResponse:
+        try:
+            return await api.get_data()
+        except InverterError as err:
+            raise UpdateFailed from err
+
+    coordinator = DataUpdateCoordinator[InverterResponse](
+        hass,
+        logger=_LOGGER,
+        name=f"solax {entry.title}",
+        update_method=_async_update,
+    )
```

Comment:
This is missing `update_interval=`. sensor.py defines a `SCAN_INTERVAL` as 30 seconds. (I intend to make it configurable in a future PR since I prefer 5...)

Reply:
adjusted in cc207e92a02359dd9b485d7cd6004297beaff97a